lib/LongVectorLoweringPass.cpp

```diff
@@ -752,6 +753,42 @@ Value *LongVectorLoweringPass::visitExtractElementInst(ExtractElementInst &I) {
   return V;
 }
 
+Value *LongVectorLoweringPass::visitGetElementPtrInst(GetElementPtrInst &I) {
+  // GEP can be tricky so we implement support only for the available test
+  // cases.
+#ifndef NDEBUG
+  {
+    assert(I.getNumIndices() == 1 &&
+           "Need test case to implement GEP with multiple indices.");
+
+    assert(I.isInBounds() && "Need test case to implement GEP not 'inbound'.");
+
+    auto *ResultTy = I.getResultElementType();
+    assert(ResultTy->isVectorTy() &&
```

Comment:
Why the restriction that the whole vector must be accessed? The code below would seem to handle accessing an individual element just as well.

Reply:
This is really down to the OpenCL test cases I had at my disposal when working on the initial prototype: I used bugpoint to reduce inputs when constructs where not supported and only this form of GEP was generated.

I suppose someone more familiar with GEPs could generate further test cases by hand to drive the improvement of this implementation.